src/features/Header/Header.jsx

```diff
@@ -33,58 +38,89 @@ function Header() {
           <b>netflix</b>
           roulette
         </a>
-        <div onClick={toggleModal}>
-          <AddMovie name={toUpper('Add Movie')} />
+        <div>
+          {movie === null ? (
+            <AddMovie
+              onClick={() => setIsOpen(true)}
+              name={toUpper('Add Movie')}
```

Comment:
create values outside of the jsx, otherwise every time you re-render they will be recreated for no reason
useMemo will help to memorise it

Reply:
Exactly, will do now